ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/V2/LedgerSeq.hs

```diff
@@ -81,7 +81,12 @@ data LedgerTablesHandle m l = LedgerTablesHandle {
     -- | Costly read all operation, not to be used in Consensus but only in
     -- snapshot-converter executable.
   , readAll            :: !(m (LedgerTables l ValuesMK))
-  , pushDiffs          :: !(LedgerTables l DiffMK -> m ())
+    -- | Push some diffs into the ledger tables handle.
+    --
+    -- The first argument has to be the ledger state before applying the block,
+    -- so that it might be in the era before the second ledger state. See
+    -- 'CanUpgradeLedgerTables'.
+  , pushDiffs          :: !(forall mk. l mk -> l DiffMK -> m ())
```

Comment:
This is slightly wrong. We should instead provide the tables outside of the ledgerstate so that we save one conversion from cardano ledger tables to unary tables. Or perhaps we are doing the same work here?